clustintime/Visualization.py

```diff
@@ -1,152 +1,251 @@
 #!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 """
-Created on Mon Sep 27 09:39:05 2021
-
-@author: ctobias
+Result visualization for clustintime
 """
 
+import matplotlib.pyplot as plt  # For graphs
+import networkx as nx  # creation, manipulation and study of the structure, dynamics and functions of complex networks
+
 # Libraries
 import numpy as np
-import matplotlib.pyplot as plt # For graphs
 import pandas as pd
-from scipy.signal import find_peaks
-import networkx as nx # creation, manipulation and study of the structure, dynamics and functions of complex networks
+import seaborn as sns
+from dyneusr import DyNeuGraph
+from dyneusr.mapper.utils import optimize_dbscan
 from IPython.display import HTML, display
-
-from kmapper import KeplerMapper, Cover
-from sklearn.decomposition import PCA 
+from kmapper import Cover, KeplerMapper
+from sklearn.decomposition import PCA
 from sklearn.manifold import TSNE
 from umap.umap_ import UMAP
-from dyneusr.mapper.utils import optimize_dbscan
-from dyneusr import DyNeuGraph
 
 
 def display_table(data):
+    """
+    HTML code for showing a table of the results
+
+    Parameters
+    ----------
+    data : numpy array
+    Returns
+    -------
+    None.
+
+    """
     html = "<table>"
     for row in data:
         html += "<tr>"
         for field in row:
-            html += "<td>%s<td>"%(field)
+            html += "<td>%s<td>" % (field)
         html += "</tr>"
     html += "</table>"
     display(HTML(html))
 
 
-def plot_labels(labels, title, multi_tap = [], single_tap = [], TR = 0.5):
-    
-    plt.figure(figsize= [32,16])
-    plt.xlim([0, len(labels)*TR])
-    plt.ylim([0,labels.max()])
-    plt.xlabel('Time in seconds', fontsize = 30)
-    plt.ylabel('# Cluster', fontsize = 30)
-    
+def plot_labels(labels, title, task=[], TR=0.5):
+    """
+    Visualization of all the clusters found along time
+
+    Parameters
+    ----------
+    labels : numpy array
+        Array of assigned clusters.
+    title : str
+        Title for the plot.
+    task : dictionary or list, optional
+        Structure containing the times when the task is performed. The default is [].
+    TR : float, optional
+        TR of the data. The default is 0.5.
+
+    Returns
+    -------
+    None.
+
+    """
+
+    colors = sns.color_palette("bright", len(task))
+
+    plt.figure(figsize=[32, 16])
+    plt.xlim([0, len(labels) * TR])
+    plt.ylim([0, labels.max()])
+    plt.xlabel("Time in seconds", fontsize=30)
+    plt.ylabel("# Cluster", fontsize=30)
+
+    for i in range(int(labels.max())):
+        selected_labels = np.array([0] * len(labels))
+        selected_labels[np.where(labels == i + 1)] = labels[np.where(labels == i + 1)]
+        plt.fill_between(np.linspace(0, len(labels) * TR, len(labels)), selected_labels)
+
+    for i in range(len(task)):
+        plt.vlines(task[i], 0, labels.max(), linewidth=1.2, colors=colors[i])
+
+
+def plot_labels_2(labels, title, task=[], TR=0.5):
```

Comment:
It is not a good idea to name the functions `plot_labels` and `plot_labels_2`.  You should use more descriptive names.